doc/user/content/concepts/hydration.md

```diff
@@ -0,0 +1,146 @@
+[sources](/concepts/sources/), [materialized
+views](/concepts/views/#materialized-views), and [indexes](/concepts/indexes/)
+in Materialize. Hydration in Materialize occurs via two specific processes, the
+snapshotting process and the hydration process.
+
+Snapshotting and hydration processes refers to the process of populating
+[sources](/concepts/sources/), [materialized
+views](/concepts/views/#materialized-views), and [indexes](/concepts/indexes/)
+in Materialize:
+
+- **Snapshotting**: Occurs when you create or recreate a source, and refers
+  to the process of populating a source in Materialize by reading the [upstream
+  (i.e., external) source's](/concepts/sources/) history and catching up to a
+  chosen offset.
+
+- **Hydration**: Occurs during cluster restarts (such as during [cluster
+  resizing](/sql/alter-cluster/#resizing) or Materialize's [weekly
+  upgrades](/releases/#schedule)), and refers to the process of populating
+  [sources](/concepts/sources/), [materialized
+  views](/concepts/views/#materialized-views), and [indexes](/concepts/indexes/)
+  from persisted data in Materialize.
+
+{{< annotation type="Disambiguation" >}}
+
+Hydration can refer to either the overarching term or the specific process of
+populating objects in Materialize. For the remainder of this page, hydration
+refers to the specific process.
+
+{{</ annotation>}}
+
+## Snapshotting
+
+Snapshotting occurs when you create or recreate a source. Snapshotting refers to
+initial population of a source in Materialize by reading the [upstream (i.e.,
+external) source](/concepts/sources/).
+
+### Process
+
+Materialize chooses the latest available offset upstream. For the latest
+available offset, Materialize uses:
+
+- Log Sequence Number (LSN) for [PostgreSQL](/ingest-data/postgres/) sources
+
+- Global Transaction Identifier (GTID) for [MySQL](/ingest-data/mysql/) sources
+
+- Kafka offset for Kafka sources
+
+Materialize then captures/commits a complete snapshot of all historical data up
+to that point. All records in this initial load have the same timestamp.
+
+Once Materialize has committed the snapshot, the source transitions from a
+**Snapshotting** status to a **Running** status, and Materialize resumes
+real-time ingestion from that offset.
+
+### Snapshotting duration and status
+
+Generally, the snapshotting duration is proportional to data volume. However,
+for sources with upsert envelope, the snapshotting may take longer while the
+upsert envelope effectively dedupes your records.
+
+You can view the snapshotting status of an object in the [Materialize
+console](/console/). For an object, go to its **Overview** page in the [Database
+object explorer](/console/data/).
+
+Alternatively, you can query the following system catalog views,
+[`mz_hydration_statuses`](/sql/system-catalog/mz_internal/#mz_hydration_statuses)
+and
+[`mz_compute_hydration_statuses`](/sql/system-catalog/mz_internal/#mz_compute_hydration_statuses).
```

Comment:
I wasn't sure about mz_internal tables, so I just briefly mention them.  But, we do include the actual sql in the Ingest data pages. Can do so here as well.